Pomodoro.Api/Middleware/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,53 @@
+﻿// <copyright file="ExceptionMiddleware.cs" company="PomodoroGroup_GL_BaseCamp">
+// Copyright (c) PomodoroGroup_GL_BaseCamp. All rights reserved.
+// </copyright>
+
+using System.Net;
+using Pomodoro.Api.Contracts;
+
+namespace Pomodoro.Api.Middleware
+{
+    /// <summary>
+    /// Represents a class that globaly handles errors.
+    /// </summary>
+    public class ExceptionMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionMiddleware"/> class.
+        /// </summary>
+        /// <param name="next">Function that can process HTTP request.</param>
+        public ExceptionMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        /// <summary>
+        /// Handles errors in case their occurence.
+        /// </summary>
+        /// <param name="context">Information about HTTP request.</param>
+        /// <returns>A <see cref="Task"/> object that represents an asynchronous operation.</returns>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await this.next(context);
+            }
+            catch (Exception e)
+            {
+                context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+
+                var error = new Error
+                {
+                    StatusCode = context.Response.StatusCode.ToString(),
+                    Message = e.Message,
+                };
+
+                context.Response.ContentType = "application/json";
+
+                await context.Response.WriteAsync(error.ToString());
```

Comment:
here I would suggest changing a little - returning ObjectResult or smth similar which has Status code in it, so you won't need to modify manually the response. 
++ in this case do not forget to return the statement on 35th line